### ISSUE TITLE
Amplify console not creating new service role for CodeCommit repo with read-only access, defaults to existing service role from other app

### ISSUE THREAD

**Issue by nbolland** [aws-amplify/amplify-hosting#2539]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-console/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-console/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-console/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

arn:aws:amplify:us-west-2:762280826255:apps/d25cg2ihyq6k5w

### Region

us-west-2

### Amplify Console feature

Not applicable

### Describe the bug

Attempting to deploy a web app from CodeCommit using the Amplify Console. I select create a new service role with read-access on the build settings page, everything goes forward as normal but fails every time on build with the following error:

2022-01-24T22:34:46.139Z [INFO]: fatal: unable to access 'https://git-codecommit.us-west-2.amazonaws.com/v1/repos/REDACTED/': The requested URL returned error: 403
2022-01-24T22:34:46.143Z [ERROR]: !!! CustomerError: Unable to clone repository due to user error code: 128

Amplify Console is unable to clone the CodeCommit repository due to the 403 error. When I check the app settings I see that instead of creating a new service role with read-only access instead it is trying to use a service role from a completely separate Amplify app.

### Expected behavior

Expected Amplify Console to create a new read-only service role for the CodeCommit repo because that's the option I selected on the settings page.

### Reproduction steps

Select "Create new service role" on the app settings page during setup.

**Comment by hloriana** [comment]
Hi @nbolland  👋🏽 thank you for raising this issue. We are looking into this.

**Comment by nbolland** [comment]
Thanks @hloriana , very much appreciated. I've been able to replicate it several times with other CodeCommit repos (obviously it's related specifically to CodeCommit) and each time I replicate it it does the same thing, defaults to an existing service role I'm currently using for a separate Amplify app from a CodeCommit repo in the same region even though I'm selecting create a new service role.

Thanks in advance and let me know if there's anything I can do.

**Comment by hloriana** [comment]
Hi @nbolland I was able to successfully create a new role when hosting my app in CodeCommit and did not come across this behavior. When you create this new role, what policies are you adding? Also if you could provide a screen shot of the role used being different from the role you created and selected, that would be helpful. Thanks!

**Comment by nbolland** [comment]
Hi @hloriana thanks for looking into it, I just gave it another try and I'm still seeing the same behavior. To clarify, I'm not manually creating the new service role, I'm just selecting the create new service role with read-only permissions option on the configure build settings page and then going through as normal, however it fails and then I look at the build settings page and I see there was not a new service role created but instead it has attached an existing service role that was created for a previous amplify app/codecommit repo.

I took some screengrabs and I'll walk you through what my process is:

1st I select create a new app, codecommit as the provider, and then I select the codecommit repo and master branch from the dropdown:
![1](https://user-images.githubusercontent.com/91922060/152885131-7755aa95-ada6-40a5-bf70-a87838e44834.png)

Then on the build settings page I leave everything set to the defaults, check the option for "Read-only access to your repository with a new service role" and check the allow amplify to to auto-deploy all files:
![2](https://user-images.githubusercontent.com/91922060/152885452-90714fe8-bde0-4969-8c4b-4782e2f6445b.png)
 
At that point I review all of the settings on the review page and click the save & deploy button:
![3](https://user-images.githubusercontent.com/91922060/152885621-d16c1f45-087e-42e8-ace8-4aeaf5abc848.png)

At this point I usually step outside to rip a heater and think about how easy my life is going to be moving forward because I no longer need to manage an s3 bucket, cloudfront dist, ACM, etc...just the repo and amplify. Then I come back and see the build fails and I get the following error during the build process:
![4](https://user-images.githubusercontent.com/91922060/152885814-59741bb9-eca2-4a88-8239-d3ed2cd6c02e.png)

At this point I'm usually sad because this has worked as expected for me before...so I head on over to the general settings page and that's when I see that the service role that has been attached is not a new service role, but in fact it's a service role that was created for a previously existing amplify app/codecommit repo that is also under the same account/region:
![5](https://user-images.githubusercontent.com/91922060/152886124-4fa7ef6b-102c-487a-a4b1-507c8778a1af.png)

Am I missing something here? When I've used amplify for this type of deploy before it has created a service role with read-only permissions for just the specific codecommit repo specified on the build settings page. What I was expecting was for the same behavior, but instead of doing that it takes an existing service role that is tied directly to a previous app/repo and assigns that as the service role.

Thanks in advance and if there's anything else I can do just let me know.

**Comment by nbolland** [comment]
Hi @hloriana - just wanted to give you an update on this issue. I went ahead and created a new service role manually for this amplify app/code commit repo and it worked fine once I had manually assigned that new service role. 

Here is the interesting part though: I then went to migrate a few other code commit repos to amplify, and when I went through the way I was doing it before and selected the "create a new service role with read-only permissions" -- this is what it was failing to create before -- now it DOES create the new service role with read-only permissions for each specific code commit repo.

I've been able to replicate this multiple times now. So, in a nutshell, 1st amplify app worked as expected when selecting "create a new service role with read-only permissions". When I tried to do that a second time it would fail and automatically assign the previous service role that was created for the 1st amplify app. After manually creating and assigning that service role, the 3rd, 4th, etc... amplify app works just fine creating that new service role.

1st one works as expected.
2nd one fails and assigns previously used service role.
3rd, 4th, and all subsequent ones work as expected.

Hope that helps. If you ever have any questions or need clarification on the issue don't hesitate to reach out.
Thanks

**Comment by hloriana** [comment]
Hi @nbolland thank you for providing these details. So is this feature working as expected for you now?

**Comment by nbolland** [comment]
Hi @hloriana. Yes, it is working as expected now. To recap:
The first codecommit repo I migrated to amplify worked exactly as expected when creating a new read-only service role.

The issue was with the 2nd codecommit repo I tried to migrate -- it would not create a new read-only service role and instead kept defaulting to the previously created read-only service role which ended up obviously erroring out every time on build.

I manually created a service role and attached it to allow it to work properly. Since doing that all subsequent codecommit repos I've migrated have worked exactly as expected and are able to create that read-only service role without issue for all subsequent codecommit repos.
